modules/azure-firewall/subnets.tf

```diff
@@ -0,0 +1,19 @@
+resource "azurerm_subnet" "main" {
+  count                = length(var.firewall)
+  name                 = "AzureFirewallSubnet"
+  resource_group_name  = azurerm_resource_group.main[count.index].name
+  virtual_network_name = azurerm_virtual_network.main[count.index].name
+  address_prefix       = "${cidrsubnet(lookup(element(var.firewall, count.index), "hub_vnet"), 2, 2)}"
+}
+
+resource "azurerm_public_ip" "main" {
+  count               = length(var.firewall)
+  name                = "aksFW-${lookup(element(var.firewall, count.index), "location")}-${var.env}-pip"
```

Comment:
the public IP may not have anything to do with AKS? 
I wouldn't hardcode it, it could be part of some name variable?